pkg/client/group_mgmt_client.go

```diff
@@ -384,3 +405,62 @@ func unwrapError(body []byte) (string, error) {
 	}
 	return errResp, nil
 }
+
+// processAsyncResponse: process http code 202 response
+func processAsyncResponse(client *GroupMgmtClient, body []byte) (interface{}, error) {
+	errResp, _ := unwrapError(body)
+	if client.WaitOnJob { // check sync flag
+		unwrapMessage := &ErrorResponse{}
+		err := json.Unmarshal(body, unwrapMessage)
+		if err != nil {
+			return nil, err
+		}
+		ids, err := handleAsyncJob(unwrapMessage, client)
+
+		if err != nil {
+			return nil, fmt.Errorf("http response error: status (202), messages: %v", err.Error())
+		}
+		return ids, fmt.Errorf("http response error: status (202), messages: %v", errResp)
+	}
+	return nil, fmt.Errorf("http response error: status (202), messages: %v", errResp)
+}
+
+//handleAsyncJob : extract the job_id and monitor it periodically until job completion or timed out
+func handleAsyncJob(resp *ErrorResponse, client *GroupMgmtClient) (string, error) {
+	// filter list of async job id from the message
+
+	var objectId string
```

Comment:
Based on my other comments below, maybe we only need this variable in scope at line 455.

Reply:
Yes.